Code/Framework/AzToolsFramework/Tests/Script/LuaEditorSystemComponentTests.cpp

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzToolsFramework/Script/LuaEditorSystemComponent.h>
+
+#include <AzCore/Script/ScriptContext.h>
+#include <AzTest/AzTest.h>
+
+namespace UnitTest
+{
+    class LuaEditorSystemComponentTests
+        : public ::testing::Test
+    {
+    public:
+        void SetUp() {}
+        void TearDown() {}
+
+        void TestValidateLuaComponentBoilerplate()
+        {
+            AZ::ScriptContext scriptContext = AZ::ScriptContext();
+            AZStd::string luaComponentBoilerplate =
+                AzToolsFramework::Script::LuaEditorSystemComponent::GenerateLuaComponentBoilerplate("TEST_SCRIPT");
+            ASSERT_TRUE(scriptContext.Execute(luaComponentBoilerplate.c_str()));
+        }
+    };
+
+    TEST_F(LuaEditorSystemComponentTests, LuaEditorSystemComponent_ValidateLuaComponentBoilerplate)
```

Comment:
Uber nit: [technically, underscores in `gtest` fixture/test names are advised against](https://github.com/google/googletest/blob/main/docs/faq.md) (I know we do this a lot already elsewhere so just drawing your attention to it 😉)

Reply:
This is awesome, thanks. Fixed in the next commit.